MAUI/DataGrid/filtering.md

```diff
@@ -0,0 +1,184 @@
+
+{% tabs %}
+{% highlight c# %}
+public bool FilterRecords(object record)
+{
+    OrderInfo orderInfo = record as OrderInfo;
+
+    if (orderInfo != null)
+    {
+        if (columns.SelectedItem.ToString() == "All Columns")
+        {
+            if (conditions.SelectedItem.ToString() == "Contains")
+            {
+                var filterText = FilterText.ToLower();
+                if (orderInfo.OrderID.ToString().ToLower().Contains(filterText) ||
+                    orderInfo.CustomerID.ToLower().Contains(filterText) ||
+                    orderInfo.Customer.ToLower().Contains(filterText) ||
+                    orderInfo.ShipCountry.ToLower().Contains(filterText) ||
+                    orderInfo.ShipCity.ToLower().Contains(filterText))
+                    return true;
+                return false;
+            }
+            else if (conditions.SelectedItem.ToString() == "Equals")
+            {
+                if (CheckEquals(orderInfo.OrderID.ToString()) ||
+                    CheckEquals(orderInfo.CustomerID) ||
+                    CheckEquals(orderInfo.Customer) ||
+                    CheckEquals(orderInfo.ShipCountry) ||
+                    CheckEquals(orderInfo.ShipCity))
+                    return true;
+                return false;
+            }
+            else
+            {
+                if (!CheckEquals(orderInfo.OrderID.ToString()) ||
+                   !CheckEquals(orderInfo.CustomerID) ||
+                   !CheckEquals(orderInfo.Customer) ||
+                   !CheckEquals(orderInfo.ShipCountry) ||
+                   !CheckEquals(orderInfo.ShipCity))
+                    return true;
+                return false;
+            }
+        }
+        else
+        {
+            var value = record.GetType().GetProperty(columns.SelectedItem.ToString().Replace(" ",""));
+            var exactValue = value.GetValue(record, null);
+            if (conditions.SelectedItem.ToString() == "Contains")
+            {
+                return FilterText.ToLower().Contains(exactValue.ToString().ToLower());
+            }
+            else if (conditions.SelectedItem.ToString() == "Equals")
+            {
+                return CheckEquals(exactValue.ToString());
+            }
+            else
+            {
+                return !CheckEquals(exactValue.ToString());
+            }
+        }
+    }
+    return false;
+}
+
+public bool CheckEquals(string value)
+{
+    return FilterText.Equals(value);
+}
+
+private void Button_Clicked(object sender, EventArgs e)
+{
+	this.sfDataGrid.View.Filter = FilterRecords;
+	this.sfDataGrid.View.RefreshFilter();
+}
+{% endhighlight %}
+{% endtabs %}
+
+The following code example illustrates how to create a [Picker](https://docs.microsoft.com/en-us/dotnet/maui/user-interface/controls/picker) for conditions and add appropriate strings to that Picker and how the records will be filtered based on selected conditions,

```

Comment:
The following code example illustrates how to create a [Picker](https://docs.microsoft.com/en-us/dotnet/maui/user-interface/controls/picker) for conditions and add appropriate strings to that Picker and how the records will be filtered based on selected conditions.

Reply:
Modified